### ISSUE TITLE
Update files accumulating in /tmp

### ISSUE THREAD

**Issue by habovh** [Jackett/Jackett#11566]
### Environment

**OS**: Linux RN316 4.4.190.x86_64.1 #1 SMP Thu Oct 22 04:41:26 UTC 2020 x86_64 GNU/Linux

### Description

Jackett does not delete files it stores in `/tmp` during the update process.
This eventually leads to update files accumulating on the disk, and in my case actually filling the disk up to 100% if no action is taken.

On low-storage devices this can become an issue, and NAS while having a lot of space on their data mount points often have tiny internal storage. I think it is safe to assume a lot of people using Jackett actually run in onto their NAS directly.

I'd like to keep Jackett up to date, so I'm not keen on disabling updates, but I think it would be a good thing to remove the previously downloaded update files after a successful update.

### Screenshots

- Listing `/tmp` folder:
![Screenshot 2021-04-19 at 15 21 27](https://user-images.githubusercontent.com/4186230/115243665-755eb380-a123-11eb-9ac2-575b88c2e0ba.png)

- Disk usage **before** removing `/tmp/JackettUpdate*` folders:
![Screenshot 2021-04-19 at 15 21 41](https://user-images.githubusercontent.com/4186230/115243674-77287700-a123-11eb-84e8-ccc33e9a879a.png)

- Disk usage **after** removing `/tmp/JackettUpdate*` folders:
![Screenshot 2021-04-19 at 15 21 54](https://user-images.githubusercontent.com/4186230/115243683-78f23a80-a123-11eb-93fb-88d80f3875a1.png)

**Comment by ilike2burnthing** [comment]
@NinjaLikesCheez / @XYZJR / @KaiSforza if any of you could have a look at implementing this it'd be a great help, thanks.

**Comment by garfield69** [comment]
It might be a case of poor logic design. the UpdateService has a CleanupTempDir routine,
https://github.com/Jackett/Jackett/blob/6a213c6eabd0264392c48564ff16a9035fd13ca4/src/Jackett.Common/Services/UpdateService.cs#L172-L200

which gets called daily from IUpdateService.cs
https://github.com/Jackett/Jackett/blob/6a213c6eabd0264392c48564ff16a9035fd13ca4/src/Jackett.Common/Services/Interfaces/IUpdateService.cs#L3-L9

But its possible that the clean-up routine may only get called if CheckForUpdatesNow() does not have an update to perform, since if there is an update control is passed to the updater task which shuts the service down ... 

Don't know that this is the case for sure, I'm not a C# dev and have not delved deeply into the Jackett code flow, but by pointing this out it may provide a volunteer a place to start investigating ;-)

**Comment by KaiSforza** [comment]
@habovh  Would be really useful to see the logs for this, if there are any.

**Comment by garfield69** [comment]
On my win10 Jackett just now, I clicked on my dashboard's **check for updates** button, and after the notification popped up that Jacket was updated, I clicked on the  **view log** button and seen
```
2021-04-20 19:06:52 | Info | Jackett startup finished in 32.271 s
2021-04-20 19:06:52 | Info | Deleting JackettUpdate temp files from C:\WINDOWS\TEMP\JackettUpdate-v0.17.908-637545423701964288
2021-04-20 19:06:52 | Info | Adding aggregate indexer ('all' indexer) ...
```
So I can confirm that on windows at least, the **CleanupTempDir()** routine is called and works OK.
Checking my **C:\WINDOWS\TEMP\\** folder there are no **JackettUpdate-\*** folders about.

So the problem may be restricted to linux platforms... perhaps it is a permissions issue?

@habovh please check your log on the dashboard after an auto-update or a manual dashboard check-for-updates, and once the update is complete, view the log and check if you get the **Deleting** message as I do, and if there are any additional related error messages post them here. Thanks.

**Comment by garfield69** [comment]
Tested this on my Win10 VirtualBox ubuntu 18.04 image, and the log shows
```
2021-04-20 20:08:05 | Info | Jackett startup finished in 3.179 s
2021-04-20 20:08:05 | Info | Deleting JackettUpdate temp files from /tmp/JackettUpdate-v0.17.908-637545460728672666
2021-04-20 20:08:05 | Info | Adding aggregate indexer ('all' indexer) ...
```
and once again there are no **JackettUpdate-\*** folders on my **/tmp/** folder

This is looking like a machine specific issue... is the OP's install of Jackett not using the standard builds or something?

**Comment by garfield69** [comment]
Tested this on my win10 VirtualBox BigSur OSx image, and all looks good there too.
So that completes the main tri-facta of Jackett standard builds.

**Comment by habovh** [comment]
Thanks for the quick reaction guys!

I might have found the issue, and it does not seem to be related to Jackett itself, but more a docs issue.

I'm using `systemd` to manage Jackett and I used the [provided systemd service from the wiki](https://github.com/Jackett/Jackett/wiki/systemd-service) to get started. Back then, I was running Jackett with mono just like in the service from the wiki.

At some point I've updated the service to use `jackett_launcher.sh` as per [this comment](https://github.com/Jackett/Jackett/issues/5208#issuecomment-547565515), but I did not change anything else than the `ExecStart` line.

I *think* that the `Restart=always` and `RestartSec=2` were trying to restart Jackett at a time it should not. I changed for `Restart=on-failure` and removed the `RestartSec` & `TimeoutStopSec` entries.

It's looking promising, but I'll know for sure after an actual update is available (tomorrow) to confirm.

If it turns out this approach for working with systemd fixes my issue I'll recommend to update the wiki or might even do so myself if that's ok.

**Comment by garfield69** [comment]
yes, please update the wiki, some of those pages are woefully out of date.
just remember that we still make available a mono flavour of jackett (despite our best efforts to drop support for it) because some niche platforms will not support the .net suite, so don't delete the mono references, just make it clear what works for which. Thanks.

**Comment by KaiSforza** [comment]
> I _think_ that the `Restart=always` and `RestartSec=2` were trying to restart Jackett at a time it should not. I changed for `Restart=on-failure` and removed the `RestartSec` & `TimeoutStopSec` entries.

I don't think this will cause an issue with the files staying there. `RestartSec` and `TimeoutStopSec` just handle things after the service has exited, and the update shouldn't cause the whole service to die. 

It could be that the updater runs, and something on your system is not allowing it to continue, but the systemd service still restarts it and it technically updates (something happens [here](https://github.com/Jackett/Jackett/blob/6a213c6eabd0264392c48564ff16a9035fd13ca4/src/Jackett.Server/Services/ServerService.cs#L339-L343)) but the changes made to the systemd service file won't make that any better.

**Comment by diegargon** [comment]
if it's good for anything, i never had that problem with Ubuntu
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.2 LTS
Release:        20.04
Codename:       focal

**Comment by habovh** [comment]
I updated the [wiki page](https://github.com/Jackett/Jackett/wiki/Systemd-service) as discussed.

**Comment by garfield69** [comment]
@habovh  Thank you.

**Comment by arf20** [comment]
I still have this problem in v0.22

![Image](https://github.com/user-attachments/assets/3c885b57-e78b-4d3e-b25d-801f03a7f07b)

**Comment by ilike2burnthing** [comment]
You could try creating a backup of your Jackett settings and indexer files, uninstall Jackett, delete any leftover files, download an installer from last week, install and then try auto-updating, see if the issue continues. Could be a permissions issue. If the problem continues, open a new issue, providing the information requested and linking to this one.
